src/SelfService/Infrastructure/Api/Kafka/KafkaTopicController.cs

```diff
@@ -216,6 +221,75 @@ public async Task<IActionResult> DeleteTopic(string id)
         return NoContent();
     }
 
+    [HttpGet("{id:required}/consumers")]
+    [ProducesResponseType(typeof(KafkaTopicApiResource), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized, "application/problem+json")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound, "application/problem+json")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError, "application/problem+json")]
+
+    public async Task<IActionResult> GetConsumers(string id)
+    {
+
+        if (!User.TryGetUserId(out var userId))
+        {
+            return Unauthorized(new ProblemDetails
+            {
+                Title = "Access denied",
+                Detail = $"User is unknown to the system."
+            });
+        }
+
+        if (!KafkaTopicId.TryParse(id, out var kafkaTopicId))
+        {
+            return NotFound(new ProblemDetails
+            {
+                Title = "Topic not found",
+                Detail = $"Topic with id \"{id}\" could not be found."
+            });
+        }
+
+        var topic = await _kafkaTopicRepository.FindBy(kafkaTopicId);
+        if (topic is null)
+        {
+            return NotFound(new ProblemDetails
+            {
+                Title = "Topic not found",
+                Detail = $"Topic with id \"{id}\" could not be found."
+            });
+        }

```

Comment:
Consider having different error messages for the two different issues. Just makes it easier to debug if we have problems here.